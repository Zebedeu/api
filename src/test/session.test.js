const createHttpServer  = require('../server');
const agent = require('supertest');
const  omit  = require('lodash');

const httpServer = createHttpServer();
const request = agent(httpServer);

afterAll(done => {
  httpServer.close(done);
});

// const User  = require('../../src/app/models/User');

describe('Authentication', () => {
    it('register', async() => {
        // const { body, status } = await global.testRequest.post('/api/user/register');
        const response = request
        .post('/api/user/register')
        .send({
            name: 'Marcio Zebedeu',
            email: 'b898@marciozebedeu.com',
            password: '1234e'
        });
        console.log(response);
        response.header = omit(response.header, ["date"]);
        expect(response).toMatchSnapshot();
    
    });
    
    it('Soudld if user is auth', async() => {
        const response = request
        .post('/api/user/login')
        .send({
            email: 'bb@marciozebedeu.com',
            password: '1234e'
        });
        // expect(response).toEqual(200);
    });
});