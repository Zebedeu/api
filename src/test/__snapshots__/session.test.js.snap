// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication register 1`] = `
Object {
  "data": Object {
    "email": "b898@marciozebedeu.com",
    "name": "Marcio Zebedeu",
    "password": "1234e",
  },
  "headers": Object {
    "content-type": "application/json",
  },
  "method": "POST",
  "url": "http://127.0.0.1:3000/api/user/register",
}
`;
